cloud/endagaweb/tasks.py

```diff
@@ -213,9 +214,23 @@ def update_credit(self, imsi, update_id):
         if request.status_code >= 200 and request.status_code < 300:
             print "update_credit SUCCESS. id=%s, imsi=%s, amount=%s. (%d)" % (
                 update_id, imsi, update.amount, request.status_code)
-            update.delete()
-            bts.mark_active()
-            bts.save()
+            with transaction.atomic():
+                expiry_date = update.valid_through
```

Comment:
This expiry date can be computed here using a denomination lookup and current subscriber balance expiry date rather than storing it in the `PendingCreditUpdate`